File: Solutions/003/Solution.py
if __name__ == "__main__":
    greatestPF, num = 0, 600851475143  

    while True:
        x = 1
        while True:
            x += 1
            if num % x == 0:
                factor = num / x
                greatestPF = x if x > greatestPF else greatestPF
                num = factor
                break
        if num == 1:
            break
        
    print("Ans: " + str(greatestPF))

File: Solutions/003/Solution.java
public class Solution{
    public static void main(String[] args){
        long num = 600851475143L;
        int greatestPF = 1;

        while(true){
            int x = 1;
            while(true){
                x += 1;
                if(num % x == 0){
                    long factor = num / x;
                    if(greatestPF < x){greatestPF = x;}
                    num = factor;
                    break;
                }
            }
            if(num == 1){break;}
        }
        System.out.println("Ans: " + greatestPF);
    }
}